core/pom.xml

```diff
@@ -30,6 +30,10 @@
          <artifactId>jgroups</artifactId>
       </dependency>
 
+      <dependency>
```

Comment:
The other better alternative than adding a compile dep to the core would, as you mentioned, a separate module like ```infinispan-kubernetes``` with the example XML and the dependencies.